### ISSUE TITLE
magicbane-0.2.0 build failure with http-conduit-2.3

### ISSUE THREAD

**Issue by DanBurton** [myfreeweb/magicbane#9]
As seen on the Stackage build server:

```
magicbane: BuildFailureException Process exited with ExitFailure 1: ./Setup build

[6 of 9] Compiling Magicbane.HTTPClient ( library/Magicbane/HTTPClient.hs, dist/
build/Magicbane/HTTPClient.o )

library/Magicbane/HTTPClient.hs:69:25: error:
    • Could not deduce (unliftio-core-0.1.1.0:Control.Monad.IO.Unlift.MonadUnlif
tIO
                          μ)
        arising from a use of ‘HCC.withResponse’
      from the context: (MonadHTTP ψ μ, MonadCatch μ)
        bound by the type signature for:
                   performWithFn :: forall ψ (μ :: * -> *) ι ρ.
                                    (MonadHTTP ψ μ, MonadCatch μ) =>
                                    (ConduitM ι ByteString μ () -> μ ρ)
                                    -> Request -> ExceptT Text μ (Response ρ)
        at library/Magicbane/HTTPClient.hs:67:1-122
      Possible fix:
        add (unliftio-core-0.1.1.0:Control.Monad.IO.Unlift.MonadUnliftIO
               μ) to the context of
          the type signature for:
            performWithFn :: forall ψ (μ :: * -> *) ι ρ.
                             (MonadHTTP ψ μ, MonadCatch μ) =>
                             (ConduitM ι ByteString μ () -> μ ρ)
                             -> Request -> ExceptT Text μ (Response ρ)
    • In the expression: HCC.withResponse req
      In the second argument of ‘($)’, namely
        ‘HCC.withResponse req
           $ \ res
               -> do body <- fn $ responseBody res
                     return res {responseBody = body}’
      In the second argument of ‘($)’, namely
        ‘tryAny
           $ HCC.withResponse req
               $ \ res
                   -> do body <- fn $ responseBody res
                         return res {responseBody = body}’
   |
69 |   res ← lift $ tryAny $ HCC.withResponse req $ \res → do
   |                         ^^^^^^^^^^^^^^^^^^^^
```

I was able to reproduce locally like so:

```bash
stack unpack magicbane-0.2.0 && cd magicbane-0.2.0
edit stack.yaml # add the following stack.yaml
stack build
```

```yaml
# stack.yaml
resolver: nightly-2018-03-10
extra-deps:
- servant-0.13
- servant-server-0.13
- http-types-0.12.1
- http-conduit-2.3.0
```

**Comment by DanBurton** [comment]
I've removed magicbane from the Stackage nightly build plan, and unfortunately it won't appear in the initial LTS 11 release.

To get back into nightly, open a PR on the Stackage repo as usual. To get into LTS 11, open an issue on https://github.com/fpco/lts-haskell/issues.

**Comment by akhra** [comment]
I think the only thing still missing here is a version bump and a hackage distribution?

**Comment by valpackett** [comment]
Yeah. I think I'll also do #11 first. (hmm actually, rio is not released yet…)

**Comment by jkachmar** [comment]
It looks like this has been resolved (at least as of LTS 12.14 `magicbane` is on Stackage again).

Can this issue be closed?
